cli/lakeflags/flags.go

```diff
@@ -33,3 +63,48 @@ func (f *Flags) HEAD() (*lakeparse.Commitish, error) {
 	}
 	return c, nil
 }
+
+func (l *Flags) Connection() (*client.Connection, error) {
+	uri, err := l.URI()
+	if err != nil {
+		return nil, err
+	}
+	if !api.IsLakeService(uri.String()) {
+		return nil, errors.New("cannot open connection on local lake")
+	}
+	conn := client.NewConnectionTo(uri.String())
+	if err := conn.SetAuthStore(l.AuthStore()); err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+func (l *Flags) Open(ctx context.Context) (api.Interface, error) {
+	uri, err := l.URI()
+	if err != nil {
+		return nil, err
+	}
+	if api.IsLakeService(uri.String()) {
+		conn, err := l.Connection()
+		if err != nil {
+			return nil, err
+		}
+		return api.NewRemoteLake(conn), nil
+	}
+	return api.OpenLocalLake(ctx, uri.String())
+}
+
+func (l *Flags) AuthStore() *auth0.Store {
+	return auth0.NewStore(filepath.Join(l.ConfigDir, credsFileName))
```

Comment:
Nit: Get rid of credsFileName.
```suggestion
	return auth0.NewStore(filepath.Join(l.ConfigDir, "credentials.json"))
```